python/paddle/fluid/dygraph/dygraph_to_static/logging_utils.py

```diff
@@ -0,0 +1,225 @@
+            cls._instance._initialized = False
+        return cls._instance
+
+    def __init__(self):
+        if self._initialized:
+            return
+
+        self._initialized = True
+        self._logger = log_helper.get_logger(
+            __name__, 1, fmt='%(asctime)s-%(levelname)s: %(message)s')
+        self._verbosity_level = None
+        self._transformed_code_level = None
+
+    @property
+    def logger(self):
+        return self._logger
+
+    @property
+    def verbosity_level(self):
+        if self._verbosity_level is not None:
+            return self._verbosity_level
+        else:
+            return int(os.getenv(VERBOSITY_ENV_NAME, DEFAULT_VERBOSITY))
+
+    @verbosity_level.setter
+    def verbosity_level(self, level):
+        self.check_level(level, {})
+        self._verbosity_level = level
+
+    @property
+    def transformed_code_level(self):
+        if self._transformed_code_level is not None:
+            return self._transformed_code_level
+        else:
+            return -1
+
+    @transformed_code_level.setter
+    def transformed_code_level(self, level):
+        self.check_level(level, _transformer_name_to_level)
+        self._transformed_code_level = level
+
+    def check_level(self, level, name_to_level_dict):
+        if isinstance(level, (six.integer_types, type(None))):
+            rv = level
+        elif str(level) == level:
+            if level not in name_to_level_dict:
+                raise ValueError("Unknown level: %r" % level)
+            rv = name_to_level_dict[level]
+        else:
+            raise TypeError("Level not an integer or a valid string: %r" %
+                            level)
+        return rv
+
+    def has_code_level(self, level):
+        level = self.check_level(level, _transformer_name_to_level)
+        return level == self.transformed_code_level
+
+    def has_verbosity(self, level):
+        level = self.check_level(level, {})
+        return level >= self.verbosity_level
+
+    def error(self, msg, *args, **kwargs):
+        self.logger.error(msg, *args, **kwargs)
+
+    def warn(self, msg, *args, **kwargs):
+        self.logger.warn(msg, *args, **kwargs)
+
+    def log(self, level, msg, *args, **kwargs):
+        if self.has_verbosity(level):
+            self.logger.log(level, msg, *args, **kwargs)
+
+    def log_transformed_code(self, level, ast_node, *args, **kwargs):
+        if self.has_code_level(level):
+            msg = ast_to_source_code(ast_node)
+            msg = "Transformed code: \n" + msg
+            self.logger.info(msg, *args, **kwargs)
+
+
+_TRANSLATOR_LOGGER = TranslatorLogger()
+
+BasicApiTransformer = 1
+TensorShapeTransformer = 2
+ListTransformer = 3
+BreakContinueTransformer = 4
+ReturnTransformer = 5
+LogicalTransformer = 6
+LoopTransformer = 7
+IfElseTransformer = 8
+AssertTransformer = 9
+PrintTransformer = 10
+CallTransformer = 11
+CastTransformer = 12
+AllTransformer = 12
+
+_transformer_name_to_level = {
```

Comment:
Can we remove these code? If we develop new Transformer and forget to add these map/integer, we made error. I comment at line 117 "def log_transformed_code" to say the implementation I thought.

Reply:
done